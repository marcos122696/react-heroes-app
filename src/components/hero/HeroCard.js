import React from 'react'
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { heroImages } from '../../helpers/heroImages';


export const HeroCard = ({
    id,
    superhero,
    publisher,
    alter_ego,
    first_appearance,
    characters
}) => {

    // const imgPath = `/assets/${id}.jpg`; public/assets

    return (
        <div className="col">
            <Link to={`/Hero/${id}`} style={{ textDecoration: 'none' }}>
                <div className="card bg-secondary text-white h-100">
                    <div className="row no-gutters">
                        <div className="container-img" >
                            <img src={ heroImages(`./${id}.jpg`)} className="h-100 w-100 animate__animated animate__zoomIn" alt={superhero} />
                        </div>

                        <div className="col-8 w-100">
                            <div className="card-body" >

                                <h5 className="card-title">{superhero}</h5>
                                <p className="card-text">{alter_ego}</p>

                                <p className="card-text">
                                    <small className="text-white-50">{first_appearance}</small>
                                </p>

                            </div>
                        </div>
                    </div>
                </div>
            </Link>
        </div>
    )
};

HeroCard.propTypes = {
    id: PropTypes.string.isRequired,
    superhero: PropTypes.string.isRequired,
    alter_ego: PropTypes.string.isRequired,
    first_appearance: PropTypes.string.isRequired,
};