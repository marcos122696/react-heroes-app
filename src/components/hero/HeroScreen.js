import React, { useMemo } from 'react';
import { useParams, Navigate, useNavigate } from 'react-router-dom';
import { heroImages } from '../../helpers/heroImages';
import { getHeroById } from '../../selectors/getHeroById';



export const HeroScreen = () => {

    const { heroeId } = useParams();

    const hero = useMemo( () => getHeroById(heroeId), [heroeId] );

    const { id, superhero, publisher, alter_ego, first_appearance, characters } = hero;
    
    const navigate = useNavigate();

    const  handleReturn  = () => {
        return navigate(-1);
    };

    if (!hero) {
        return <Navigate to='/' />;
    };

    // const imgPath = `/assets/${ id }.jpg`; public/assets

  return (
    <div className="row mt-5">
        <div className="col-4">
            <img 
                src={ heroImages(`./${id}.jpg`) }
                alt={ superhero }
                className="img-thumbnail animate__animated animate__fadeInLeft"
            />
        </div>

        <div className="col-8">
            <h3>{ superhero }</h3>
            <ul className="list-group list-group-flash">
                <li className="list-group-item list-group-item-primary"> <b>Alter ego:</b> { alter_ego }</li>
                <li className="list-group-item list-group-item-primary"> <b>publisher:</b> { publisher }</li>
                <li className="list-group-item list-group-item-primary"> <b>first Appearance:</b> { first_appearance }</li>
            </ul>

            <h5 className="mt-5">Characters</h5>
            <p>{ characters }</p>

            <button
                className="btn btn-outline-info"
                onClick={ handleReturn }
            >
                Regresar
            </button>

        </div>
    </div>
  )
}
