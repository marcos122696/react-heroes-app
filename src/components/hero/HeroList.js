import React, { useMemo } from 'react'
import { getHeroesByPublisher } from '../../selectors/getHeroesByPublisher'
import { HeroCard } from './HeroCard';
import PropTypes from 'prop-types';

export const HeroList = ({ publisher }) => {
    
    const heroes = useMemo( () => getHeroesByPublisher(publisher), [publisher] );

  return (
    <div className="row rows-cols-1 row-cols-md-3 g-3 text-dark">
        {
            heroes.map( (hero) => (
                <HeroCard 
                    key={ hero.id } 
                    { ...hero}
                />
            ))
        }
    </div>
  )
};

HeroList.propTypes = {
    publisher: PropTypes.string.isRequired,
};
