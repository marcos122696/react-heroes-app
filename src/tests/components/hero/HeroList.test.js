import { shallow } from "enzyme";
import { HeroList } from "../../../components/hero/HeroList";
import { heroes } from "../../../data/heroes";


describe('Pruebas en el <HeroList />', () => {

    const wrapper = shallow(
        <HeroList 
            publisher={ heroes[0].publisher }
        />
    );

    test('Debe mostrarse correctamente', () => { 
        expect( wrapper ) .toMatchSnapshot();
     })
})