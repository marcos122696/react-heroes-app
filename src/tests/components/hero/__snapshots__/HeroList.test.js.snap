// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <HeroList /> Debe mostrarse correctamente 1`] = `
<div
  className="row rows-cols-1 row-cols-md-3 g-3 text-dark"
>
  <HeroCard
    alter_ego="Bruce Wayne"
    characters="Bruce Wayne"
    first_appearance="Detective Comics #27"
    id="dc-batman"
    key="dc-batman"
    publisher="DC Comics"
    superhero="Batman"
  />
  <HeroCard
    alter_ego="Kal-El"
    characters="Kal-El"
    first_appearance="Action Comics #1"
    id="dc-superman"
    key="dc-superman"
    publisher="DC Comics"
    superhero="Superman"
  />
  <HeroCard
    alter_ego="Jay Garrick"
    characters="Jay Garrick, Barry Allen, Wally West, Bart Allen"
    first_appearance="Flash Comics #1"
    id="dc-flash"
    key="dc-flash"
    publisher="DC Comics"
    superhero="Flash"
  />
  <HeroCard
    alter_ego="Alan Scott"
    characters="Alan Scott, Hal Jordan, Guy Gardner, John Stewart, Kyle Raynor, Jade, Sinestro, Simon Baz"
    first_appearance="All-American Comics #16"
    id="dc-green"
    key="dc-green"
    publisher="DC Comics"
    superhero="Green Lantern"
  />
  <HeroCard
    alter_ego="Oliver Queen"
    characters="Oliver Queen"
    first_appearance="More Fun Comics #73"
    id="dc-arrow"
    key="dc-arrow"
    publisher="DC Comics"
    superhero="Green Arrow"
  />
  <HeroCard
    alter_ego="Princess Diana"
    characters="Princess Diana"
    first_appearance="All Star Comics #8"
    id="dc-wonder"
    key="dc-wonder"
    publisher="DC Comics"
    superhero="Wonder Woman"
  />
  <HeroCard
    alter_ego="J'onn J'onzz"
    characters="Martian Manhunter"
    first_appearance="Detective Comics #225"
    id="dc-martian"
    key="dc-martian"
    publisher="DC Comics"
    superhero="Martian Manhunter"
  />
  <HeroCard
    alter_ego="Dick Grayson"
    characters="Dick Grayson"
    first_appearance="Detective Comics #38"
    id="dc-robin"
    key="dc-robin"
    publisher="DC Comics"
    superhero="Robin/Nightwing"
  />
  <HeroCard
    alter_ego="Dan Garret"
    characters="Dan Garret, Ted Kord, Jaime Reyes"
    first_appearance="Mystery Men Comics #1"
    id="dc-blue"
    key="dc-blue"
    publisher="DC Comics"
    superhero="Blue Beetle"
  />
  <HeroCard
    alter_ego="Dinah Drake"
    characters="Dinah Drake, Dinah Lance"
    first_appearance="Flash Comics #86"
    id="dc-black"
    key="dc-black"
    publisher="DC Comics"
    superhero="Black Canary"
  />
</div>
`;
